libraries/chain/controller.cpp

```diff
@@ -2104,13 +2114,7 @@ struct controller_impl {
    }
 
    checksum256_type calculate_action_merkle() {
-      vector<digest_type> action_digests;
-      const auto& actions = pending->_block_stage.get<building_block>()._actions;
-      action_digests.reserve( actions.size() );
-      for( const auto& a : actions )
-         action_digests.emplace_back( a.digest() );
-
-      return merkle( move(action_digests) );
+      return merkle( move(pending->_block_stage.get<building_block>()._action_receipt_digests) );
```

Comment:
It seems risky to clobber _action_receipt_digests here.

Reply:
It is a bit risky. Could add a `clear()` after which would make it a bit less risky. Or could create a version of` merkle` that takes by `const&` instead of by value.

Reply:
Moved the `move` to finalize block as suggested by Arhag.